### PR TITLE
Charts: bugfixing, improvements and safety chart

### PR DESCRIPTION
**Pull request by jorgemiguelreis**
-> Div scroll on charts page is needed ❗️

### CODE REVIEW

File: src/static/js/pages/run/runStats.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ var runStats = function(element) {
             },
             type: 'GET',
             success: function(data, status) {
-                _chartsHandler(element, data.data);
+                _chartsHandler(element, data.data.charts);
```
Comment (marcosoliveira91) [inline src/static/js/pages/run/runStats.js]: what about the sessionStats?

**Comment by marcosoliveira91** [comment]
@NelsonPereira1991 pls review and merge this asap

File: src/static/js/pages/run/runStats.js (hunk 1 above)
Comment (jorgemiguelreis) [reply src/static/js/pages/run/runStats.js]: right, i didn't change the schema of the object. I will change to an object with an array of charts

Comment (jorgemiguelreis) [reply src/static/js/pages/run/runStats.js]: done 👍

File: src/controllers/stats.js [hunk 2]
```diff
@@ -1,35 +1,25 @@
...
+const path = require("path");
+const servicesFolder = global.servicesFolder;
+const statsService = require(path.resolve(servicesFolder, "stats"));
+const responseUtil = require(path.resolve(global.utilsFolder, "response"));
+
+module.exports = {
+    getRunStats: function (req, res) {
+
+        if (req.get("Content-Type") === "application/json") {
+            let data = statsService.getRunStats(req.session._doc, req.run._doc, req.iterations, req.events);
```
Comment (NelsonPereira1991) [inline src/controllers/stats.js]: 👍

File: src/services/stats.js [hunk 3]
```diff
@@ -1,167 +1,284 @@
...
+                    }
+                }]
+            }
+        }
+    };
+
+    return chart;
+}
+
+function buildBarChart(labels, datasets, title, xLabel, yLabel){
```
Comment (NelsonPereira1991) [inline src/services/stats.js]: you are not using labels for anything inside the function

File: src/services/stats.js [hunk 4]
```diff
@@ -1,167 +1,284 @@
...
+    });
+
+    let datasets = [
+        buildBarDataset([generalSafetyEvents.length], 'General Safety', colorsUtil.getRandomColor()),
+        buildBarDataset([materialDropEvents.length], 'Material Drop', colorsUtil.getRandomColor())
+    ];
+
+    labels = [1,2];
+
+    return buildBarChart(labels, datasets, 'Safety events', ''/*'Iteration'*/, 'Occurrences');
```
Comment (NelsonPereira1991) [inline src/services/stats.js]: - labels not used
- remove the commented Iteration string

File: src/services/stats.js [hunk 5]
```diff
@@ -1,167 +1,284 @@
...
+    });
+
+    let datasets = [
+        buildBarDataset([assemblyErrorEvents.length], 'Assembly Errors', colorsUtil.getRandomColor()),
+        buildBarDataset([materialErrorEvents.length], 'Material Errors', colorsUtil.getRandomColor())
+    ];
+
+    labels = [1,2];
+
+    return buildBarChart(labels, datasets, 'Quality events', ''/*'Iteration'*/, 'Occurrences');
```
Comment (NelsonPereira1991) [inline src/services/stats.js]: - labels not used
- remove the commented Iteration string

**Pull request merged**